dyno-core/src/main/java/com/netflix/dyno/connectionpool/Host.java

```diff
@@ -122,13 +122,18 @@ public InetSocketAddress getSocketAddress() {
         return socketAddress;
     }
 
+    /**
+     * The HostSupplier may not have the ability to return the port
+     * and therefore the hashCode() may fail if the user provides his
+     * own port. Hence, we do not have to include the port number in the
+     * hashCode() calculation.
+     */
     @Override
     public int hashCode() {
         final int prime = 31;
         int result = 1;
         result = prime * result + ((hostname == null) ? 0 : hostname.hashCode());
         result = prime * result + ((rack == null) ? 0 : rack.hashCode());
```

Comment:
Do we have a use case of running multiple dynomite nodes on teh same node even if it means for testing purpose?